src/main/kotlin/org/jetbrains/bio/big/Bed.kt

```diff
@@ -59,35 +59,35 @@ data class BedEntry(
     /**
      * Unpacks a basic bed3 [BedEntry] into a bedN+ [ExtendedBedEntry].
      *
-     * Correctly parses the typed BED fields (such as score, color or blockSizes). Extra (non-standard) fields
-     * are stored in [ExtendedBedEntry.extraFields] property as a [String] array.
+     * Correctly parses the typed BED fields (such as score, color or blockSizes). Extra (non-standard) fields,
+     * if any (and if [parseExtraFields] is enabled) are stored in [ExtendedBedEntry.extraFields]
+     * property as a [String] array (null if no extra fields or [parseExtraFields] is disabled).
      *
      * @throws BedEntryUnpackException if this entry couldn't be parsed, either because there are too few fields
      * or because the field values don't conform to the standard, e.g. score is not an integer number. The exception
      * contains the BED entry and the index of the offending field.
      *
      * @param fieldsNumber Expected regular BED format fields number to parse (3..12)
-     * @param extraFieldsNumber BED+ format extra fields number to parse, if null parse all extra fields
+     * @param parseExtraFields Whether to parse or discard the BED+ format extra fields.
      * @param delimiter Custom delimiter for malformed data
      * @param omitEmptyStrings Treat several consecutive separators as one
      */
     fun unpack(
             fieldsNumber: Byte = 12,
-            extraFieldsNumber: Int? = null,
+            parseExtraFields: Boolean = true,
```

Comment:
Seems it is inconsistent with `filedsNumber`. We validate bed fields number and throw exceptions if number differs from expected. On the other hand now we are "relaxed" in terms of extra fields number, any number allowed. Is it intended behavior?

Reply:
It is.
I posit that, since extra fields are extra anyway, and we give them to the user as-is, there's no point in imposing strict limitations on them. Any semantics that they might have is the user's responsibility. The regular fields are another story -- we parse them. If the user specifies `bed9`, they must really want that color field, and it would be misleading to just fill it with default value.
Another point is that within my implementation, once you know the delimiter, the set of formats becomes strictly ordered: the sets of `BedEntry`s recognizable by any two formats are nested. Any `BedEntry` can be unpacked with `bed3+`, any `bed7+` entry is simultaneously a `bed5+` entry etc. This is important because under this condition you can determine a common format for any file.

Reply:
> , since extra fields are extra anyway, and we give them to the user as-is
I don't agree.

Pure BED format is not widely used by tools, all peak callers used own fixed bedN+K format (macs, sicer, span) with fixed `k`. 

If we expected that our format is bed3+5, than we assume that we could use extra fields 0..4 and won't get ArrayOutOfBounds exception, or etc. In case of bed3+ we indeed could don't care of extra fields number

> If the user specifies bed9, they must really want that color field, and it would be misleading to just fill it with default value.

The same idea here, if the user expects 'K' extra fileds it would misleading to get AIOBE when accessing to k-th field. Why he is required to check each time that indeed `K` fileds were extraced(we still parse them to an array) if he already asked to parse `k` fields?

Reply:
See, in my current version, the user simply can't specify `bed3+5`, because there's no such thing. So it's consistent and sufficiently powerful (`bedN+K` format is a subset of `bedN+` format).
But we can revert this decision and return to the support of all three versions (`bedN+`, `bedN+K` and `bedN` as a special case of `bedN+K` with `K == 0`) if you believe that this is a viable use case. Should we throw a separate exception on missing extra fields or just let the user figure where the error lies?

Reply:
> if you believe that this is a viable use case. 

@dievsky So let's leave it as you like. My point was that there are some new inconsistency of `bedN+K` vs `bedN` parsing (relaxed for K and strict for N), I'm not sure that it is a critical or viable use case, we could return to this in future if smth goes wrong.